hdf5-rs/src/hl/space.rs

```diff
@@ -49,6 +54,69 @@ impl Deref for Dataspace {
 }
 
 impl Dataspace {
+
+    /// Create a simple dataspace covering the given `shape`
+    pub fn create_simple(shape: &[Ix]) -> Result<Dataspace> {
+        let shape_u64: Vec<u64> = shape.iter().map(|v| *v as u64).collect();
+        let id = h5try!(H5Screate_simple(shape.len() as i32, shape_u64.as_ptr(), ptr::null()));
+        Ok(Dataspace::from_handle(Handle::try_new(id)?))
+    }
+
+    /// Select a slice (known as a 'hyperslab' in HDF5 terminology) of the Dataspace.
+    /// Returns the shape of array that is capable of holding the resulting slice.
+    /// Useful when you want to read a subset of a dataset.
+    pub fn select_slice<T, D>(&self, slice: SliceInfo<T,D>) -> Result<Vec<Ix>> where
+    T: AsRef<[SliceOrIndex]>, D: ndarray::Dimension
+    {
+        let shape = self.dims();
+        let mut start_vec = Vec::new();
```

Comment:
These vectors can be allocated with the expected capacity.

```suggestion
        let mut start_vec = Vec::with_capacity(ss.len());
```